src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -97,213 +92,409 @@ Position const SpawnBat[6] =
-            events.ScheduleEvent(EVENT_MIND_FLAY, 10s, 30s, PHASE_TWO);
-            events.ScheduleEvent(EVENT_GREATER_HEAL, 25s, PHASE_TWO);
-            events.ScheduleEvent(EVENT_SPAWN_FLYING_BATS, 10s, PHASE_TWO);
+            scheduler.CancelGroup(PHASE_ONE);
 
-            return;
-        }
+            // Curse of Blood
+            scheduler.Schedule(5s, 15s, PHASE_TWO, [this](TaskContext context)
+            {
+                DoCastSelf(SPELL_CURSE_OF_BLOOD);
+                context.Repeat(25s, 30s);
+            // Psychic Scream
+            }).Schedule(25s, 35s, PHASE_TWO, [this](TaskContext context)
+            {
+                DoCastVictim(SPELL_PSYCHIC_SCREAM);
+                context.Repeat(35s, 45s);
+            // Shadow Word: Pain
+            }).Schedule(10s, 15s, PHASE_TWO, [this](TaskContext context)
+            {
+                DoCastRandomTarget(SPELL_SHADOW_WORD_PAIN, 0, true);
+                context.Repeat(12s, 18s);
+            // Mind Flay
+            }).Schedule(10s, 30s, PHASE_TWO, [this](TaskContext context)
+            {
+                DoCastVictim(SPELL_MIND_FLAY);
+                context.Repeat(20s, 40s);
+            // Greater Heal
+            }).Schedule(25s, PHASE_TWO, [this](TaskContext context)
+            {
+                Talk(EMOTE_GREAT_HEAL);
+                me->InterruptNonMeleeSpells(false);
+                DoCastSelf(SPELL_GREATER_HEAL);
+                context.Repeat(25s);
+            // Spawn Flying Bats
+            }).Schedule(10s, PHASE_TWO, [this](TaskContext context)
+            {
+                if (me->GetThreatMgr().GetThreatListSize())
+                {
+                    // summon up to 2 bat riders
+                    if (batRidersCount < 2)
+                    {
+                        LOG_DEBUG("scripts.ai", "boss_jeklik::UpdateAI:: Spawn Flying Bats (Summoning {} of 2)", batRidersCount + 1);
+                        Talk(SAY_CALL_RIDERS);
+                        // only if the bat rider was successfully created
+                        if (me->SummonCreature(NPC_BATRIDER, SpawnBatRider, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT))
+                        {
+                            // increase the counter
+                            batRidersCount++;
+                        }
+                        if (batRidersCount == 1)
+                        {
+                            context.Repeat(10s, 15s);
+                        }
+                    }
+                }
+            });
+        });
     }
 
-    void UpdateAI(uint32 diff) override
+    void EnterEvadeMode(EvadeReason why) override
     {
-        if (!UpdateVictim())
-            return;
+        // enter evade mode
+        BossAI::EnterEvadeMode(why);
 
-        events.Update(diff);
+        if (why != EvadeReason::EVADE_REASON_NO_PATH)
+        {
+            // make invisible to hide wonky-looking movement
+            me->SetVisible(false);
```

Comment:
This feels...hacky, though I don't know of a better way to correct it.

Reply:
Couldn't agree more but after a few hours of messing with movement I can't get her to run back to a ground home position before teleporting herself up to the cave. There is no path to get up there, so a smooth transition is impossible.

To add to the pain, if she evades while gliding down from her cave, she tends to get caught halfway through the spline path and I've found it nearly impossible to cancel it cleanly. It MAY be related to [#17595](https://github.com/azerothcore/azerothcore-wotlk/issues/17595), but regardless I can't make it look right.

This solution isn't too bad from the player's PoV - she does a reasonable-length fade-out, and then several seconds later appears in her home spot casting the right thing. It is the best of the options I was able to figure out how to do.

TL;DR - I tried. I failed. This doesn't suck.

Reply:
I'm going to resolve this one for now, please re-open if you feel it merits further discussion.